### DISCUSSION TITLE
[Feature Request]: Redesign the playlist view

### DISCUSSION THREAD

**Discussion by pkrasicki** [FreeTubeApp/FreeTube#6294] [General]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

1. The playlist view has two different layouts. In grid mode it's a single column layout and in list mode it's a two column layout. As far as I know, no other page behaves like this. There are also inconsistencies. The two column layout displays a thumbnail (but only above a certain resolution) and the single column layout doesn't display a thumbnail at all.

2. On mobile, videos in a playlist are always displayed as a list, no matter if we're in grid mode or in list mode. Every other page respects the setting on all resolutions.

3. The search bar is in the wrong place. It should be next to the list of videos, in the same row as the Sort By dropdown.

4. The design of this page is kinda different from all other pages, but there isn't really a good reason for that.

### Proposed Solution

1. Remove the two column layout. A single column layout can display both a grid and a list and works well on all resolutions without hacks. Having just one layout will make the code simpler and testing easier. Let's add a thumbnail to the single column layout (right now it's missing).

2. Display videos in a way that's consistent with other pages. Grid/list mode setting should always be respected.

3. Move the search bar to the other panel.

4. We could reuse the design of the channel page (or at least get as close as possible). It works well in all resolutions and users are already familiar with it.

![playlist2](https://github.com/user-attachments/assets/c7b7668a-e696-47e0-845e-7818324ea2aa)

### Alternatives Considered

Some of those problems can be fixed without a redesign, but I didn't want to create separate issues for them.

### Issue Labels

visual improvement

### Additional Information

In the design, the big block of text is playlist's description. I just wanted to show that we can fit long text. It doesn't have to be in that exact spot. The tabs are not necessary, I just wanted to show that we can have them if needed.

**Comment by kommunarr** [comment]
Thanks for documenting these! To add my two cents, I think there is merit to the problems that are mentioned, but any specific solutions are always prone to opinions and discussions. 2 & 3 are probably the easiest to solve on their own without a broader redesign, and 1 & 4 are probably where the most room for debate is.

To reference some of our team's prior discussions on 1, the current list view more closely resembles the desktop playlist view for YT and other similar apps, but the grid view is a strong favorite for many users' regular usage. So we've decided on the time being to keep that form for list view fans and the remote playlist view (where the benefits of grid view, e.g., more onscreen content visible at one time, are not as relevant). As for 4, the list view is intentionally similar to the playlist view of other apps, and the grid view is actually probably what we want the History page to be more like (i.e., convenient sticky top bar). I'm not saying these are perfect solutions, but it will take considerable development effort, testing, consensus building, compromising, and reworking to get any sizable redesign across the finish line. That's not to say it can't happen, but it will take a good deal of social effort and weighting against higher priority tasks.

**Comment by pkrasicki** [comment]
Thanks for explaining the reasons behind the current design. My reason to make it look similar to the channel page is that both pages serve the same purpose - to display a collection of videos and channel/playlist information + some action buttons. It makes sense for pages that serve a similar purpose to look the same. Just like Subscriptions, History, Trending and Most Popular look similar to each other. But I've checked and you are right that YouTube has that two column layout too (it switches to single column on mobile just like FreeTube). Not all apps do it like that, though. 

YouTube
![youtube screenshot](https://github.com/user-attachments/assets/38ff762b-9181-43fc-a171-dc99c53d0dc8)

Odysee
![odysee screenshot](https://github.com/user-attachments/assets/90feab52-e110-425c-9ab8-7e15b8278dcb)

Piped
![piped screenshot](https://github.com/user-attachments/assets/38f11ce9-3c3b-4526-8ad4-c642c57bcdc8)

FreeTube (development branch, list mode)
![freetube dev](https://github.com/user-attachments/assets/6105a43f-53dd-4026-b14e-57ffc24a0c66)
